reward_index/src/indexer.rs

```diff
@@ -120,34 +121,36 @@ impl Indexer {
     fn extract_reward_share(&self, msg: &[u8]) -> Result<(RewardKey, u64)> {
         match self.mode {
             settings::Mode::Mobile => {
-                let share = RadioRewardShare::decode(msg)?;
-                let key = RewardKey {
-                    key: share.hotspot_key,
-                    reward_type: RewardType::MobileGateway,
-                };
-                Ok((key, share.amount))
+                let share = MobileRewardShare::decode(msg)?;
+                match share.reward {
+                    Some(MobileReward::RadioReward(r)) => Ok((
+                        RewardKey {
+                            key: r.hotspot_key,
+                            reward_type: RewardType::MobileGateway,
+                        },
+                        r.dc_transfer_reward + r.poc_reward,
+                    )),
+                    _ => bail!("got an invalid mobile reward share"),
+                }
```

Comment:
```suggestion
                let Some(MobileReward::RadioReward(r)) = share.reward else {
                    bail!("got an invalid mobile reward share");
                };
                Ok((
                     RewardKey {
                         key: r.hotspot_key,
                         reward_type: RewardType::MobileGateway,
                     },
                     r.dc_transfer_reward + r.poc_reward,
                ))
```
Hope I typed this right

Reply:
I think it is better as is, since it matches the structure of the iot branch